File: PlatformUI2.0/app/src/modules/elasticsearch/service/elasticSearchService.ts
/*******************************************************************************
 * Copyright 2017 Cognizant Technology Solutions
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License.  You may obtain a copy
 * of the License at
 * 
 *   http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations under
 * the License.
 ******************************************************************************/

/// <reference path="../../../_all.ts" />

module ISightApp {
	export interface IElasticSearchService {
		loadKibanaIndex(): ng.IPromise<any>;
		queryNeo4jData(queryTerm : string) : ng.IPromise<any>;
	}

	export class ElasticSearchService implements IElasticSearchService {
		static $inject = ['$resource', '$q', '$cookies', 'restCallHandlerService', 'restEndpointService'];

		constructor(private $resource, private $q, private $cookies, private restCallHandlerService: IRestCallHandlerService, private restEndpointService: IRestEndpointService) { }

		loadKibanaIndex(): ng.IPromise<any> {
			var restHandler = this.restCallHandlerService;
            return restHandler.get("SEARCH_DASHBOARD");
		}

		queryNeo4jData(queryTerm : string) : ng.IPromise<any>{
			var elasticSearchResource = this.$resource(this.restEndpointService.getelasticSearchServiceHost()+'/neo4j-index/_search?from=0&size=100&q=*'+queryTerm+'*',
				{},
				{
					get: {
						method: 'GET',
					}
				});
			return elasticSearchResource.get().$promise;
		}
	}
}

File: PlatformService/src/main/java/com/cognizant/devops/platformservice/rest/es/ElasticSearchService.java
/*******************************************************************************
 * Copyright 2017 Cognizant Technology Solutions
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License.  You may obtain a copy
 * of the License at
 * 
 *   http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations under
 * the License.
 ******************************************************************************/
package com.cognizant.devops.platformservice.rest.es;

import javax.servlet.http.HttpServletRequest;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.cognizant.devops.platformcommons.config.ApplicationConfigProvider;
import com.cognizant.devops.platformcommons.dal.elasticsearch.ElasticSearchDBHandler;
import com.cognizant.devops.platformcommons.exception.InsightsCustomException;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonParser;
import com.google.gson.JsonSyntaxException;

@RestController
@RequestMapping("/search")
public class ElasticSearchService {
	static Logger log = LogManager.getLogger(ElasticSearchService.class.getName());

	@Autowired
	private HttpServletRequest context;

	@RequestMapping(value = "/data", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
	public String searchData(@RequestParam String query, @RequestParam(required = false, defaultValue = "0") int from,
			@RequestParam(required = false, defaultValue = "100") int size) throws JsonSyntaxException, InsightsCustomException {
		ElasticSearchDBHandler dbHandler = new ElasticSearchDBHandler();
		String url = ApplicationConfigProvider.getInstance().getEndpointData().getElasticSearchEndpoint()
				+ "/neo4j-index/_search?from=" + from + "&size=" + size + "&q=*" + query + "*";
		return new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create()
				.toJson(new JsonParser().parse(dbHandler.search(url)));
	}
}
